src/ui/map/index.js

```diff
@@ -0,0 +1,332 @@
+require('qs-hash');
+const geojsonRewind = require('geojson-rewind');
+
+const DrawRectangle = require('../draw/rectangle');
+const ExtendDrawBar = require('../draw/extend_draw_bar');
+const {
+  EditControl,
+  SaveCancelControl,
+  TrashControl
+} = require('./controls');
+const {
+  addIds,
+  addMarkers,
+  geojsonToLayer,
+  bindPopup
+} = require('./util');
+
+let writable = false;
+
+const dummyGeojson = {
+  type: 'FeatureCollection',
+  features: [
+    {
+      type: 'Feature',
+      geometry: {
+        type: 'Point',
+        coordinates: [0, 0],
+      },
+    },
+  ],
+};
+
+const DARK_FEATURE_COLOR = '#555';
+const LIGHT_FEATURE_COLOR = '#e8e8e8';
+
+module.exports = function (context, readonly) {
+  writable = !readonly;
+
+  function maybeShowEditControl() {
+    // if there are features, show the edit button
+    if (context.data.hasFeatures()) {
+      d3.select('.edit-control').style('display', 'block');
+    }
+  }
+
+  function map(selection) {
+    mapboxgl.accessToken =
+      'pk.eyJ1IjoiY2hyaXN3aG9uZ21hcGJveCIsImEiOiJjbDR5OTNyY2cxZGg1M2luejcxZmJpaG1yIn0.mUZ2xk8CLeBFotkPvPJHGg';
+
+    context.map = new mapboxgl.Map({
```

Comment:
How would you feel about assigning this to `window.map`? I frequently access the map instance directly in order to achieve things that pages don't otherwise implement.

Reply:
Great idea!

Reply:
I'll push a commit with this now.